src/test/txvalidation_tests.cpp

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2017 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <validation.h>
+#include <txmempool.h>
+#include <amount.h>
+#include <consensus/validation.h>
+#include <primitives/transaction.h>
+#include <script/script.h>
+#include <test/test_bitcoin.h>
+
+#include <boost/test/unit_test.hpp>
+
+
+BOOST_AUTO_TEST_SUITE(txvalidation_tests)
+
+/**
+ * Ensure that the mempool won't accept coinbase transactions.
+ */
+BOOST_FIXTURE_TEST_CASE(tx_mempool_reject_coinbase, TestChain100Setup)
+{
+    CScript scriptPubKey = CScript() << ToByteVector(coinbaseKey.GetPubKey()) << OP_CHECKSIG;
+    CMutableTransaction coinbaseTx;
+
+    coinbaseTx.nVersion = 1;
+    coinbaseTx.vin.resize(1);
+    coinbaseTx.vout.resize(1);
+    coinbaseTx.vin[0].scriptSig = CScript() << OP_11 << OP_EQUAL;
+    coinbaseTx.vout[0].nValue = 1 * CENT;
+    coinbaseTx.vout[0].scriptPubKey = scriptPubKey;
+
+    assert(CTransaction(coinbaseTx).IsCoinBase());
+
+    CValidationState state;
+
+    LOCK(cs_main);
+
+    unsigned int initialPoolSize = mempool.size();
+
+    BOOST_CHECK_EQUAL(
+            false,
+            AcceptToMemoryPool(mempool, state, MakeTransactionRef(coinbaseTx),
+                nullptr /* pfMissingInputs */,
+                nullptr /* plTxnReplaced */,
+                true /* bypass_limits */,
+                0 /* nAbsurdFee */));
+
+    // Check that the transaction hasn't been added to mempool.
+    BOOST_CHECK_EQUAL(mempool.size(), initialPoolSize);
+
+    // Check that the validation state reflects the unsuccesful attempt.
+    BOOST_CHECK(state.IsInvalid());
+    BOOST_CHECK_EQUAL(state.GetRejectReason(), "coinbase");
+
+    int nDoS;
+    BOOST_CHECK_EQUAL(state.IsInvalid(nDoS), true);
```

Comment:
Call this above:
```cpp
int nDoS;
BOOST_CHECK(state.IsInvalid(nDoS), true);
BOOST_CHECK_EQUAL(state.GetRejectReason(), "coinbase");
BOOST_CHECK_EQUAL(nDoS, 100);
```

Reply:
@MarcoFalke any reason to not consider the suggestion?

Reply:
@promag was happy to make the change but didn't have time yet, though it's worth noting that I'm uncertain on what the real difference would be aside from cosmetics. Note that `state.IsInvalid` with and without an argument call to separate methods, so why not get coverage on both?

Reply:
To me this seems mostly a cosmetic nit, subject to personal preference of style. For test related pulls I don't see a reason to block them solely based on style nits.